lib/v2.recommended.yaml

```diff
@@ -0,0 +1,231 @@
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# This software is derived from pedantic (https://github.com/dart-lang/pedantic)
+# with separate copyright notices and license terms:
+#
+# Copyright 2017, the Dart project authors. All rights reserved.
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+# Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+# Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following
+# disclaimer in the documentation and/or other materials provided
+# with the distribution.
+# Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived
+# from this software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Every lint in this file is categorized as "recommended".
+#
+# For context on that decision or to discuss a change for any individual lint,
+# visit this issue and search for the lint by its name:
+# https://github.com/Workiva/workiva_analysis_options/issues/2
+
+# This file also includes all of the required rules.
+include: package:workiva_analysis_options/v1.yaml
+
+analyzer:
+  errors:
+    # Promote enabled lints to errors:
+    always_declare_return_types: warning
```

Comment:
@evanweible-wf 

> RP has already been using this. It will create a bit of noise for over_react class-based components since the render method is often untyped, but with the move towards function components this becomes less of an issue. Also, the suggestion I think is to type render methods as Object.

I have never understood this line of reasoning. I frequently see those typed as ReactElement. Honestly, I don't think we should be allowing anything else. Yes, technically String is allowed but it isn't really useful on a component. I can't think of a single time I have ever done that. Null is fine, even in the future (ReactElement?).

I raise this, because this argument has always felt empty to me. The value of this lint far outweighs this one instance in my mind. This tells me that I must be missing something.

Reply:
👀 I'm stalking this PR

+1000000, I do not find this lint noisy

Reply:
I was just digging through code making sure I understood what was happening in DPC. A lot of our component render methods return dynamic. This is very functionally similar to having no return type declared but is at least explicit rather than implicit, which I prefer because I don't think a lot of developers think about how leaving the return type off doesn't mean you don't have a return.

Reply:
Sorry I don't think I worded that well – I'm not citing that as a reason against enforcing this, just noting that it may produce a lot of infractions in codebases that have class-based components and haven't previously used this lint. I talked with the FED squad yesterday and they were on board with adding this lint.

fwiw, they suggested using `Object` rather than `dynamic` because the latter will prevent the analyzer from warning about a missing return, but `ReactElement` would also work if the component isn't going to return a primitive like a String like you said.